src/java/com/zimbra/graphql/resolvers/impl/ContactResolver.java

```diff
@@ -0,0 +1,169 @@
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.IDS, description="Comma-separated item ids to act on") String ids,
+        @GraphQLArgument(name=GqlConstants.CONSTRAINT, description="Target constraints") String constraint,
+        @GraphQLRootContext RequestContext context) throws ServiceException {
+        return contactRepository.itemTrash(context, ids, constraint);
+    }
+
+    @GraphQLMutation(description="Adds the specified tags to the given contacts.")
+    public ActionResult contactTag(
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.IDS, description="Comma-separated item ids to act on") String ids,
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.TAG_NAMES, description="The tags to add") String tagNames,
+        @GraphQLArgument(name=GqlConstants.CONSTRAINT, description="Target constraints") String constraint,
+        @GraphQLRootContext RequestContext context) throws ServiceException {
+        return contactRepository.itemTag(context, ids, tagNames, constraint);
+    }
+
+    @GraphQLMutation(description="Removes the specified tags from the given contacts.")
+    public ActionResult contactUntag(
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.IDS, description="Comma-separated item ids to act on") String ids,
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.TAG_NAMES, description="The tags to remove") String tagNames,
+        @GraphQLArgument(name=GqlConstants.CONSTRAINT, description="Target constraints") String constraint,
+        @GraphQLRootContext RequestContext context) throws ServiceException {
+        return contactRepository.itemUntag(context, ids, tagNames, constraint);
+    }
+
+    @GraphQLMutation(description="Updates a given contact with the specified properties.")
+    public FolderActionResult contactUpdate(
+        @GraphQLNonNull @GraphQLArgument(name=GqlConstants.IDS, description="Comma-separated item ids to act on") String ids,
+        @GraphQLArgument(name=GqlConstants.FOLDER_ID, description="The destination folder") String folderId,
+        @GraphQLArgument(name=GqlConstants.FLAGS, description="The flags to set") String flags,
+        @GraphQLArgument(name=GqlConstants.TAG_NAMES, description="The tags to set") String tagNames,
+        @GraphQLArgument(name=GqlConstants.RGB, description="RGB color in format #rrggbb where r,g and b are hex digits") String rgb,
+        @GraphQLArgument(name=GqlConstants.COLOR, description="color numeric; range 0-127; defaults to 0 if not present; client can display only 0-7") Byte color,
+        @GraphQLArgument(name=GqlConstants.ATTRIBUTES, description="The attributes to set") List<NewContactAttr> attributes,
+        @GraphQLRootContext RequestContext context) throws ServiceException {
+        return contactRepository.contactUpdate(context, ids, folderId, flags, tagNames, rgb, color, attributes);
+    }
+
+    @GraphQLMutation(description="Modifies a given contact with the specified properties.")
```

Comment:
why 2 methods contactUpdate() and contactModify() ?
what's the difference?

Reply:
That's outlined in the comment above.